test/interface/composite_algorithm_test.jl

```diff
@@ -20,3 +20,4 @@ solve!(integrator2)
 @test integrator1.sol.t == integrator2.sol.t
 
 sol = solve(prob,alg_switch)
+@inferred OrdinaryDiffEq.DiffEqBase.__solve(prob,alg_switch)
```

Comment:
If possible (maybe it's still broken?), it might be better to test
```suggestion
@inferred DiffEqBase.__init(prob, alg_switch)
```
to check the modified function directly and ensure that not only the solution but also the integrator type can be inferred.